### ISSUE TITLE
Cannot be reproduced

### ISSUE THREAD

**Issue by Zong-Liang** [dengxl0520/MemSAM#8]
Dear author, thanks for your great work. I encountered the following issues when trying to reproduce the paper:
- Processed the CAMUS dataset according to your method.
- Modified the dataset path in the code and supplemented the class.json file.
![deb5b28b2992f6c226cf97bf4a1b6c36](https://github.com/dengxl0520/MemSAM/assets/74233357/69679c04-58c3-488a-8ce6-c104de573895)
- Other parameter configurations remain unchanged.
![9e0254b061dcdc355ac97b7f60348d60](https://github.com/dengxl0520/MemSAM/assets/74233357/d609464e-9cf6-4499-b068-6cd71f4eadc2)
![image](https://github.com/dengxl0520/MemSAM/assets/74233357/aa2b5d7b-cc23-4811-ba87-b74ae4fc9334)
However, I am unable to train normally:
![image](https://github.com/dengxl0520/MemSAM/assets/74233357/263b3bc3-ee09-4d7a-a947-cb7069e5e64f)
I'm not sure why this is happening. Could you please help me? I'd greatly appreciate your help. Thank you so much.

**Comment by dengxl0520** [comment]
You can check the inputs to the model. I'll update a few times later on how to use this training code.

**Comment by Zong-Liang** [comment]
> You can check the inputs to the model. I'll update a few times later on how to use this training code.

Thanks for your kindly reply, here is an example input to ths model:
![image](https://github.com/dengxl0520/MemSAM/assets/74233357/ae0d7501-5df6-4bd9-ba1d-f08127c4153c)

And I visualized some of the preds from the model, it's weird:
![image](https://github.com/dengxl0520/MemSAM/assets/74233357/fa4835ab-fcaf-487a-b057-39f9bd8a81f6)

I don't know where the problem is. I haven't changed anything in the model.

**Comment by xychen2022** [comment]
Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.

**Comment by SUNNYDAY-4176** [comment]
> Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.

I didn't see this file either

**Comment by Zong-Liang** [comment]
> Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.

see closed issue#6

**Comment by xychen2022** [comment]
Same problem here. The training behavior seems not normal. The training loss is low (ranges from -10000 to -2000 after a few steps). Any comments?

**Comment by songWell** [comment]
> Dear author, thanks for your great work. I encountered the following issues when trying to reproduce the paper:
> 
> * Processed the CAMUS dataset according to your method.
> * Modified the dataset path in the code and supplemented the class.json file.
>   ![deb5b28b2992f6c226cf97bf4a1b6c36](https://private-user-images.githubusercontent.com/74233357/339657100-69679c04-58c3-488a-8ce6-c104de573895.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4MzU2MTQsIm5iZiI6MTcxOTgzNTMxNCwicGF0aCI6Ii83NDIzMzM1Ny8zMzk2NTcxMDAtNjk2NzljMDQtNThjMy00ODhhLThjZTYtYzEwNGRlNTczODk1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAxVDEyMDE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiZDFkNTMxZDdjMTNhM2Q0N2YwZDVmNTkxZWI0ZjQ5MmQxZTU0YjMxMmIzN2JmOGMxYTIxYjg0NjA1NGI1NGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-MqvRllZ7O5oJi72e9xCG4oEpIrXktNXu_dy9HBoUEg)
> * Other parameter configurations remain unchanged.
>   ![9e0254b061dcdc355ac97b7f60348d60](https://private-user-images.githubusercontent.com/74233357/339658385-d609464e-9cf6-4499-b068-6cd71f4eadc2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4MzU2MTQsIm5iZiI6MTcxOTgzNTMxNCwicGF0aCI6Ii83NDIzMzM1Ny8zMzk2NTgzODUtZDYwOTQ2NGUtOWNmNi00NDk5LWIwNjgtNmNkNzFmNGVhZGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAxVDEyMDE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjZmFlMWZmNGQ5M2E5MWJkNGViMjliMmI2YjUxNTUwYTU1NjZjMTY1ZTgyMmI2YzNhODdiMDVkODViZjFlNTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OUfa4WjKp1XCggVK0nOhwhhPxxIk6rli2bQ9_RYKL7c)
>   ![image](https://private-user-images.githubusercontent.com/74233357/339658575-aa2b5d7b-cc23-4811-ba87-b74ae4fc9334.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4MzU2MTQsIm5iZiI6MTcxOTgzNTMxNCwicGF0aCI6Ii83NDIzMzM1Ny8zMzk2NTg1NzUtYWEyYjVkN2ItY2MyMy00ODExLWJhODctYjc0YWU0ZmM5MzM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAxVDEyMDE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4YjEyNzc2ZTBmN2FkZDlhOTQ1MDI0M2NhMDZmODgyZjk1YWQxM2E4N2QwYjk2NTk1ZmRhMmVmZDEwMGQ5OWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NZ7fxfc4wI9hyj6jpnz4K0wuWT8x1FbY9_BI24T6Rkw)
>   However, I am unable to train normally:
>   ![image](https://private-user-images.githubusercontent.com/74233357/339658183-263b3bc3-ee09-4d7a-a947-cb7069e5e64f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk4MzU2MTQsIm5iZiI6MTcxOTgzNTMxNCwicGF0aCI6Ii83NDIzMzM1Ny8zMzk2NTgxODMtMjYzYjNiYzMtZWUwOS00ZDdhLWE5NDctY2I3MDY5ZTVlNjRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAxVDEyMDE1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI0N2NiYjUyODk3NGE5MjJmN2FiOTFjYjYzODI4ZTMxNDdhNzExNjcxZGMwZDFlNmMzYmZjZTRkNDEyYzk1MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.S2asFlRVzwny2D50ndBBg6LyolrU7HXMsYD1IaoUoQQ)
>   I'm not sure why this is happening. Could you please help me? I'd greatly appreciate your help. Thank you so much.

Same Problem
Should it be added some parameters ?  python train_video.py --enable_memory --reinforce. 
The parameters information printed are False

**Comment by songWell** [comment]
@dengxl0520 
hi, I am new in this field so I don't know if I have made any mistakes. Is it correct to use this command "python train_video.py --enable_memory --reinforce" to train the model. After training, I test the model, but the output is all zero. Can you give me some suggestions?

@Zong-Liang
 Hello, have you successfully reproduced it?

**Comment by Baldwinzc** [comment]
> > Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> 
> see closed issue#6 参见已关闭的 issue#6

The issue labeled "issue 6" has been closed. Could you please inform me how to proceed with this?

**Comment by Zong-Liang** [comment]
> > > Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> 
> > 
> 
> > see closed issue#6 参见已关闭的 issue#6
> 
> 
> 
> The issue labeled "issue 6" has been closed. Could you please inform me how to proceed with this?
> 
> 

it's from the SAMUS repo，you can refer to it，but the class.json is not the key problem here， i've abandoned reproduction（My abilities are not enough），good luck to U.

**Comment by Zong-Liang** [comment]
> @dengxl0520 
> 
> hi, I am new in this field so I don't know if I have made any mistakes. Is it correct to use this command "python train_video.py --enable_memory --reinforce" to train the model. After training, I test the model, but the output is all zero. Can you give me some suggestions?
> 
> 
> 
> @Zong-Liang
> 
>  Hello, have you successfully reproduced it?
> 
> 

No，i've abandoned，wasted too much time here～

**Comment by nhtsymh** [comment]
> > > class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有file.class.json的情况下运行 train_video.py 时，我遇到了 FileNotFound 错误 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> > 
> > 
> > See closed issue#6 参见已关闭的 issue#6
> 
> 标记为“问题 6”的问题已关闭。您能告诉我如何进行吗？

Hello, have you resolved the issue with the `class.json`? I'm encountering the same problem.

**Comment by Baldwinzc** [comment]
Thank you very much for your answer

> > > > Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> > 
> > 
> > > 
> > 
> > 
> > > see closed issue#6 参见已关闭的 issue#6
> > 
> > 
> > The issue labeled "issue 6" has been closed. Could you please inform me how to proceed with this?
> 
> it's from the SAMUS repo，you can refer to it，but the class.json is not the key problem here， i've abandoned reproduction（My abilities are not enough），good luck to U.

Thank you very much for your answer

**Comment by Baldwinzc** [comment]
![5aa0c00cb391a15ae6454e8802a2f7a](https://github.com/user-attachments/assets/dc2465a5-679b-498d-9fa4-bcaae242bf85)
I encountered the same problem

**Comment by wuyu-sile** [comment]
> > > > Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> > 
> > 
> > > 
> > 
> > 
> > > see closed issue#6 参见已关闭的 issue#6
> > 
> > 
> > The issue labeled "issue 6" has been closed. Could you please inform me how to proceed with this?
> 
> it's from the SAMUS repo，you can refer to it，but the class.json is not the key problem here， i've abandoned reproduction（My abilities are not enough），good luck to U.

我已经解决该问题了，猜测原因是作者未给出专门用于处理camus数据集的代码部分，因此导致camus的mask张量中出现255，正确的mask张量中的元素应该是非0即1的，因此需要在data_us文件中的第501行加入代码：mask[mask == 255] = 1。

**Comment by wuyu-sile** [comment]
> > > > Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> > 
> > 
> > > 
> > 
> > 
> > > see closed issue#6 参见已关闭的 issue#6
> > 
> > 
> > The issue labeled "issue 6" has been closed. Could you please inform me how to proceed with this?
> 
> it's from the SAMUS repo，you can refer to it，but the class.json is not the key problem here， i've abandoned reproduction（My abilities are not enough），good luck to U.

在data_us文件中的第501行加入mask[mask == 255] = 1之后，训练结束后结果如下，可惜作者没给出处理camus数据集的预处理代码部分，看不到实际的训练结果
![memsam](https://github.com/user-attachments/assets/bb10c598-c236-4480-86b3-0f34cf1c2302)

**Comment by Baldwinzc** [comment]
> > > > > Is class.json automatically generated in preprocessing? I didn't see such a file after running the code for preprocessing. And I got FileNotFound error when running train_video.py without the file.class.json 是否在预处理中自动生成？运行代码进行预处理后，我没有看到这样的文件。在没有文件的情况下运行 train_video.py 时，我收到了 FileNotFound 错误。
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > see closed issue#6 参见已关闭的 issue#6
> > > 
> > > 
> > > The issue labeled "issue 6" has been closed. Could you please inform me how to proceed with this?
> > 
> > 
> > it's from the SAMUS repo，you can refer to it，but the class.json is not the key problem here， i've abandoned reproduction（My abilities are not enough），good luck to U.
> 
> 在data_us文件中的第501行加入mask[mask == 255] = 1之后，训练结束后结果如下，可惜作者没给出处理camus数据集的预处理代码部分，看不到实际的训练结果 ![memsam](https://private-user-images.githubusercontent.com/178457766/368846454-bb10c598-c236-4480-86b3-0f34cf1c2302.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3MzMxODgsIm5iZiI6MTcyNjczMjg4OCwicGF0aCI6Ii8xNzg0NTc3NjYvMzY4ODQ2NDU0LWJiMTBjNTk4LWMyMzYtNDQ4MC04NmIzLTBmMzRjZjFjMjMwMi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxOVQwODAxMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYmI0ODI0NGE0YjJkZjNhN2QxMmM1MmVhYTY2Zjg0N2M2ZmExYWVjMTJkYjkzYzk5NTcwMDE4NmFjMGQ1MWE4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.G2e6twDqo4S8Aq2GxHiUrzGMilZCU0kvb8Glm5Px0uw)

I am very grateful for the time you took out of your busy schedule to respond to my issue. Your solution was extremely helpful, and I have successfully replicated the project following your guidance. Your expertise and willingness to help are truly appreciated.

Thank you once again for your valuable time and assistance; it has been very meaningful to me. If there is any way I can return the favor or assist you in the future, please do not hesitate to let me know.
